### ISSUE TITLE
AIDriver.lua:679: attempt to index field 'course' (a nil value)

### ISSUE THREAD

**Issue by MADxHAWK** [Courseplay/courseplay#4217]
I have a silo compactor running in "filling up" mode and a CP course tipping backward into the silo. 
Usualy the unloading vehicle stopped when approching the silo and waited untill the compacting vehicle left the silo and reached its parkposition. 
Now the unloading vehicle slows down but dont stop and continue approching the silo what most times causes a crash betwenn the silo leaving compacting vehicle and the unloading vehicle. 
After the crash i noticed the error 
AIDriver.lua:679: attempt to index field 'course' (a nil value)
in the logfile

This might be regarding report #4198 too 

What logoptions i should turn on for a detailed log? 

CP-Version: v6.01.00295

**Comment by Tensuko** [comment]
What Mode did the Tracktor run ? Let me guess... Mode2 ?

**Comment by MADxHAWK** [comment]
Yes, since it cant drive through i set it to tip backwards

**Comment by Flashsolutions** [comment]
Completely different scenario but produced the same error.  My issue was closed as duplicate but in an earlier post of the problem I was asked to reproduce it with debug set and resubmit, which I did  today and  posted the log.   And yes, mode 2.  This keeps happening.  I can reproduce it.   Can not a nil check be made?  
[log.txt](https://github.com/Courseplay/courseplay/files/3638981/log.txt)
